lite2/client.go

```diff
@@ -561,15 +561,38 @@ func (c *Client) verifyHeader(newHeader *types.SignedHeader, newVals *types.Vali
 			panic(fmt.Sprintf("Unknown verification mode: %b", c.verificationMode))
 		}
 	} else {
-		// 2) Otherwise, perform backwards verification
-		// Find the closest trusted header after newHeader.Height
-		var closestHeader *types.SignedHeader
-		closestHeader, err = c.trustedStore.SignedHeaderAfter(newHeader.Height)
+		var firstHeaderHeight int64
+		firstHeaderHeight, err = c.FirstTrustedHeight()
 		if err != nil {
-			return errors.Wrapf(err, "can't get signed header after height %d", newHeader.Height)
+			return errors.Wrapf(err, "can't get sig")
+		}
+		var closestHeader *types.SignedHeader
+		if newHeader.Height < firstHeaderHeight {
+			closestHeader, err = c.TrustedHeader(firstHeaderHeight)
+			if err != nil {
+				return errors.Wrapf(err, "can't get signed header after height %d", newHeader.Height)
+			}
+			if HeaderExpired(closestHeader, c.trustingPeriod, now) {
+				closestHeader = c.latestTrustedHeader
+			}
+			err = c.backwards(closestHeader, newHeader, now)
+		} else {
+			closestHeader, err = c.trustedStore.SignedHeaderBefore(newHeader.Height)
+			if err != nil {
+				return errors.Wrapf(err, "can't get signed header before height %d", newHeader.Height)
+			}
+			var closestValidatorSet *types.ValidatorSet
+			if HeaderExpired(closestHeader, c.trustingPeriod, now) {
+				closestHeader = c.latestTrustedHeader
```

Comment:
not sure this will work... 

say firstHeaderHeight = 10
newHeaderHeight = 20
lastTrustedHeader = 30

if firstHeaderHeight expired, we'll try to bisect from 30 to 20.

Reply:
complexity creates errors

Reply:
right. I'm just going to use this:
```
if HeaderExpired(closestHeader, c.trustingPeriod, now) {
	err = c.backwards(c.latestTrustedHeader, newHeader, now)
}
```